python/Ganga/GPIDev/Lib/File/OutputFileManager.py

```diff
@@ -76,19 +73,17 @@ def getOutputSandboxPatterns(job):
 
     return outputPatterns
 
-"""
-we have to set the inputsandbox patterns for the input files that will be copied from the client, also write the commands for downloading input files from the WN
-"""
-
-
 def getInputFilesPatterns(job):
+    """
+    we have to set the inputsandbox patterns for the input files that will be copied from the client, also write the commands for downloading input files from the WN
+    """
 
     tmpDir = tempfile.mkdtemp()
 
     inputPatterns = []
 
     # if GangaDataset is used, check if they want the inputfiles transferred
-    inputfiles_list = copy.deepcopy(job.inputfiles)
+    inputfiles_list = job.inputfiles if job.inputfiles else []
```

Comment:
Am I right in saying that because this is a `GangaList` the deep copy is implicit? I ask because of the:
```
inputfiles_list += job.inputdata.files
```
below.